core/services/ardupilot_manager/firmware_download/FirmwareDownload.py

```diff
@@ -299,12 +300,16 @@ def download(self, vehicle: Vehicle, platform: Platform, version: str = "") -> O
 
         path = FirmwareDownload._download(url)
 
+        if not path:
+            logger.error("Failed to download firmware.")
+            return None
+
         firmware_format = FirmwareDownload._supported_firmware_formats[platform]
         if firmware_format == FirmwareFormat.ELF:
             # Make the binary executable
```

Comment:
can we get a better comment here? I can infer what those S_ flags are, but I would rather not...

Reply:
Done.